### PR TITLE
dev-python/parver: Introducing a new package

### PR DESCRIPTION
**Pull request by bucefal91**
New package being introduced, dev-python/parver in the latest available version.

Package successfully emerges and passes the tests.

Closes: https://bugs.gentoo.org/669266
Signed-off-by: Oleksandr Trotsenko <oleksandr.trotsenko@gmail.com>
Package-Manager: Portage-2.3.51, Repoman-2.3.11

### CODE REVIEW

File: dev-python/parver/parver-0.1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+
+DEPEND="
+	dev-python/arpeggio[${PYTHON_USEDEP}]
+	dev-python/setuptools[${PYTHON_USEDEP}]
+	test? (
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		<dev-python/pytest-4.0.0[${PYTHON_USEDEP}]
```
Comment (mgorny) [inline dev-python/parver/parver-0.1.1.ebuild]: Why those `<` deps? That's just asking to make Python team's life PITA in the future.

File: dev-python/parver/parver-0.1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		<dev-python/pretend-2.0.0[${PYTHON_USEDEP}]
+	)
+"
+
+python_test() {
+	py.test -v || die "Testing failed"
```
Comment (mgorny) [inline dev-python/parver/parver-0.1.1.ebuild]: `pytest -vv`

File: dev-python/parver/parver-0.1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python2_7 python3_{4,5,6} )
```
Comment (mgorny) [inline dev-python/parver/parver-0.1.1.ebuild]: 3.7? I guess some deps are missing it (arpeggio?); I'd appreciate if you could test them and include in this (or some other) PR if they test fine on 3.7. If they're missing tests, adding python_test would be even better.

**Comment by bucefal91** [comment]
@mgorny hi! :)

I see, I've removed the `<` dependencies. I literally copied the dependencies from the upstream. Next time I'll make sure I do not restrain it on non-existing yet packages.

You are right, it's a dependency for net-proxy/mitmproxy-4.0.4. The mitmproxy bug is https://bugs.gentoo.org/572170 Other dependencies for MITM proxy are:
* https://github.com/gentoo/gentoo/pull/10197
* https://github.com/gentoo/gentoo/pull/10196

Eventually it was `arpeggio` & `memory_profiler` who were lacking Python 3.7 support. So I amended my commit in this pull request to bump versions of those packages and to include support for the latest python within them.

I just wonder if such practice is OK within gentoo? I see those packages have some maintainers.. would they get offended/upset if some wimp like me comes in and makes himself comfortable within _their_ packages?

**Comment by bucefal91** [comment]
Right... If I sat there for at least 30 seconds questioning whether to amend or to add a new commit, I would have chosen to append a new commit... but I didn't sit there thinking.

I choke it up into 3 commits. And next time I'll try to visit the dependencies to include python3_7 or things of that nature, even if they have some maintainer declared. :)

File: dev-python/arpeggio/arpeggio-1.9.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+PYTHON_COMPAT=( python2_7 python3_{4,5,6,7} )
+
+inherit distutils-r1
+
+MY_PN=Arpeggio
+MY_P=${MY_PN}-${PV}
+DESCRIPTION="Parser interpreter based on PEG grammars"
+HOMEPAGE="https://pypi.org/project/Arpeggio/ https://github.com/igordejanovic/Arpeggio"
+SRC_URI="https://github.com/igordejanovic/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+LICENSE="MIT"
```
Comment (mgorny) [inline dev-python/arpeggio/arpeggio-1.9.0.ebuild]: Empty line above LICENSE would be nice.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Copyright 1999-2018 Gentoo Authors
```
Comment (mgorny) [inline dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: Version missing in commit message.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+inherit distutils-r1
+
+DESCRIPTION="A module for monitoring memory usage of a python program"
+HOMEPAGE="https://pypi.org/project/memory_profiler/ https://github.com/fabianp/memory_profiler"
+SRC_URI="mirror://pypi/${P:0:1}/${PN}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="test"
```
Comment (mgorny) [inline dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: I don't see it user anywhere.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+	${EPYTHON} -m memory_profiler test/test_func.py || die
+	${EPYTHON} -m memory_profiler test/test_loop.py || die
+	${EPYTHON} -m memory_profiler test/test_as.py || die
+	${EPYTHON} -m memory_profiler test/test_global.py || die
+	${EPYTHON} -m memory_profiler test/test_precision_command_line.py || die
+	${EPYTHON} -m memory_profiler test/test_gen.py || die
+	if python_is_python3; then
+		${EPYTHON} -m memory_profiler test/test_unicode.py || die
+	fi
+	${PYTHON} test/test_tracemalloc.py || die
```
Comment (mgorny) [inline dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: Why PYTHON?

**Comment by mgorny** [comment]
@zmedico, ok to bump arpeggio?

File: dev-python/arpeggio/arpeggio-1.9.0.ebuild (hunk 4 above)
Comment (bucefal91) [reply dev-python/arpeggio/arpeggio-1.9.0.ebuild]: Empty line inserted.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild (hunk 5 above)
Comment (bucefal91) [reply dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: Added the exact version in the commit message.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild (hunk 6 above)
Comment (bucefal91) [reply dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: I've corrected this one. I took the least intrusive approach - to copy-paste old ebuild and luckily it has built so I didn't see much inside.

File: dev-python/memory_profiler/memory_profiler-0.54.0.ebuild (hunk 7 above)
Comment (bucefal91) [reply dev-python/memory_profiler/memory_profiler-0.54.0.ebuild]: Same as above, I didn't alter the pre-existing ebuild as long as it has built. After consulting the difference between `$PYTHON` and `$EPYTHON` I understand your point and have changed the variable :)

**Comment by zmedico** [comment]
> @zmedico, ok to bump arpeggio?

Yes, I've merged thee version bumps in a7a395375041869ce3e602b62f077ac285a0d63c and be901b2db524390c1be7bb85185d449145e4e2c8.

**Comment by mgorny** [comment]
Rebase, please.

**Comment by bucefal91** [comment]
I've rebased and kept only the parver commit since others got 'ported' into master by @zmedico

**Comment by mgorny** [comment]
```
commit 2722b2b8ad6f9bde8816123da74e7ba66ba6628a
Author:     Oleksandr Trotsenko <oleksandr.trotsenko@gmail.com>
AuthorDate: Sun Oct 28 04:59:13 2018
Commit:     Michał Górny <mgorny@gentoo.org>
CommitDate: Sat Nov 3 09:22:38 2018

    dev-python/parver: Introducing a new package
    
    New package being introduced, dev-python/parver in the latest available version
    of 0.1.1.
    
    Closes: https://bugs.gentoo.org/669266
    Signed-off-by: Oleksandr Trotsenko <oleksandr.trotsenko@gmail.com>
    Package-Manager: Portage-2.3.51, Repoman-2.3.11
    Signed-off-by: Michał Górny <mgorny@gentoo.org>
```

Sorry, failed at tags.

**Pull request closed without merging**